extension/src/validator/validator-builder.js

```diff
@@ -6,8 +6,21 @@ function withPayment(paymentObject) {
   const errors = {}
 
   return {
+    validateMetadataFields() {
+      if (!paymentObject.custom && !paymentObject.custom.fields) return this
+      if (!paymentObject.custom.fields.commercetoolsProjectKey ||
+          paymentObject.custom.fields.commercetoolsProjectKey === ' ')
+        errors.missingRequiredCtpProjectKey =
+            errorMessages.MISSING_REQUIRED_FIELDS_CTP_PROJECT_KEY
+      if (!paymentObject.custom.fields.adyenMerchantAccount ||
+          paymentObject.custom.fields.adyenMerchantAccount === ' ')
+        errors.missingRequiredAdyenMerchantAcc =
+            errorMessages.MISSING_REQUIRED_FIELDS_ADYEN_MERCHANT_ACCOUNT
+      return this
+    },
     validateRequestFields() {
-      if (!paymentObject.custom) return this
+      if (!paymentObject.custom || errors.missingRequiredCtpProjectKey || errors.missingRequiredAdyenMerchantAcc)
```

Comment:
I don't think this method needs to check `missingRequiredCtpProjectKey` and `missingRequiredAdyenMerchantAcc`. It checks only fields with `Request` ending, so it should not care about other fields.

Reply:
9f61a99